src/plugins/query_enhancements/public/datasets/s3_type.ts

```diff
@@ -171,9 +174,13 @@ const fetchConnections = async (
   http: HttpSetup,
   dataSource: DataStructure
 ): Promise<DataStructure[]> => {
-  const query = (dataSource.meta as DataStructureCustomMeta).query;
-  const response = await http.fetch(`../../api/enhancements/datasource/external`, {
-    query,
+  const abortController = new AbortController();
```

Comment:
What do we do with this abort controller?

Reply:
need to verify if using it correctly yet. but in the case of the user selecting a connection accidentally and then selecting another connection immediately aborting the previous call